### ISSUE TITLE
Option --connection-token-file not functional in "code serve-web"

### ISSUE THREAD

**Issue by jgoodson** [microsoft/vscode#215537]
Does this issue occur when all extensions are disabled?: Yes

- VS Code Version: 1.90.1
- OS Version: Rocky 8

Steps to Reproduce:

1. Run `code serve-web`  with the `--connection-token` option with a string containing the desired connection token.
2. The server starts up with the specified connection token.
3. Run `code serve-web`  with the `--connection-token-file` option with a path to a file with the desired connection token.
4. The server starts up with a randomly-generated UUID as the connection token.

The `code serve-web` exposes a `--connection-token-file` which would reasonably be expected to work, especially since `--connection-token` _does_ work as expected and overrides generation of a random token. In particular, in [serve_web.rs](https://github.com/microsoft/vscode/blob/dc8c2d019c806a06aa47a4e7a6bd147b5d68b27e/cli/src/commands/serve_web.rs#L77) `serve_web()` calls `mint_connection_token()` to handle either generating a fresh token or using a provided one. This only handles `args.connection_token` and as far as I can tell there is no code in the option handling that would replace this value with the token read from the file `args.connection_token_file`. The `--connection-token-file` option only appears to be used in different contexts like running the downloaded servers, and the value in the ServeWebArgs version appears to only ever be overwritten by the previously-mentioned token minting code.

When executing the `code serve-web` function, it would be much preferable to provide a specified connection token in a file, for reasons similar to those in https://github.com/microsoft/vscode/issues/207491 to prevent the connection token from being accessible by other users. I do not believe this is a security vulnerability, since you have to explicitly opt in to providing a connection token in a potentially unsafe manner as-is, but being able to provide one in a file would be a substantial security improvement when using specified connection tokens. I believe that this `serve-web` code should read in the file at the path provided by the `--connection-token-file` argument and use the contents as the token when provided. If this isn't intended to be used, I would think it should be removed from the options provided for the CLI `serve-web` command instead of being non-functional.

**Comment by eleanorjboyd** [comment]
Hi @jgoodson! Are you able to verify if this fix works? You can do so by getting the latest pre-release version of the remote-ssh extension and trying these same steps there. We want to make sure we get this bug verified, and the fix is correct, before we release this to stable. Thanks!

**Comment by jgoodson** [comment]
I checked the code-insiders build I downloaded today (1.92.0). It does not appear to have a fix for this issue and the current insiders build has the same behavior as before.

I also took a look at the [PR](https://github.com/microsoft/vscode/pull/219041) that closed this issue and although it is titled as relevant to this issue, it contains a [single commit](https://github.com/microsoft/vscode/pull/219041/commits/4c0517e1409deae7bf9568af673caab628c61f2e) that appears like it would perhaps be relevant to the remote-ssh extension but does not have any relation to this issue with connection tokens and the CLI that I can tell. I think perhaps that PR/commit got mixed up from different issue than this one.
